tests/datasets.py

```diff
@@ -0,0 +1,62 @@
+from dataclasses import dataclass
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+from sknnr.datasets import load_moscow_stjoes
+
+
+@dataclass
+class TestDataset:
+    __test__ = False
+
+    X_train: np.ndarray
+    X_test: np.ndarray
+    y_train: np.ndarray
+    y_test: np.ndarray
+    ref_distances: np.ndarray
+    ref_neighbors: np.ndarray
+    trg_distances: np.ndarray
+    trg_neighbors: np.ndarray
+    trg_predicted_weighted: np.ndarray
+    trg_predicted_unweighted: np.ndarray
```

Comment:
I have a lot to learn about typing, but can you explain why you type these as `np.ndarray` whereas in `datasets._base`, you use
```
from numpy.typing import NDArray
ids: Union[NDArray[np.int64], "pd.DataFrame"]
```

Reply:
Good catch, that was just inconsistency on my part! I think I should be using `NDArray` here as well.

I *think* the only difference between the two (explained [here](https://numpy.org/devdocs/reference/typing.html#ndarray)) is that `NDArray` enforces a type so that a type checker would catch the edge case of mutating the dtype. Probably not something that will ever come up, but we should at least stay consistent.